services/src/main/java/org/keycloak/services/managers/ApplianceBootstrap.java

```diff
@@ -127,4 +141,31 @@ public void createMasterRealmUser(String username, String password) {
         ServicesLogger.LOGGER.addUserSuccess(username, Config.getAdminRealm());
     }
 
+    public void createTemporaryMasterRealmAdminService(String clientId, String clientSecret /*, Integer expriationMinutes*/) {
+        RealmModel realm = session.realms().getRealmByName(Config.getAdminRealm());
+        session.getContext().setRealm(realm);
+
+        clientId = StringUtil.isBlank(clientId) ? DEFAULT_TEMP_ADMIN_SERVICE : clientId;
+        //expriationMinutes = expriationMinutes == null ? DEFAULT_TEMP_ADMIN_EXPIRATION : expriationMinutes;
+
+        ClientModel adminClient = session.clients().addClient(realm, clientId);
+        adminClient.setEnabled(true);
+        adminClient.setServiceAccountsEnabled(true);
+        adminClient.setSecret(clientSecret);
+        adminClient.setProtocol(OIDCLoginProtocol.LOGIN_PROTOCOL);
+        adminClient.setClientAuthenticatorType(KeycloakModelUtils.AUTH_TYPE_CLIENT_SECRET);
+        
+        new ClientManager(new RealmManager(session)).enableServiceAccount(adminClient);
+        UserModel serviceAccount = session.users().getServiceAccount(adminClient);
+        RoleModel adminRole = realm.getRole(AdminRoles.ADMIN);
+        serviceAccount.grantRole(adminRole);
+
```

Comment:
@vmuzikar @mabartos any idea why this is insufficient for the service account to work as an admin?

Reply:
You mean the role is correctly assigned (e.g. visible in the UI) but the service account still doesn't have admin permissions? That's weird as it works for a normal admin user.

Possibly @mposolda @pedroigor might have some hints?

Reply:
I see the service account roles have admin assigned, but using the service account lacks authorization.

Reply:
@shawkins There seem to be some compilation error. I tried at least with main – assigned admin role to service account and used it in kcadm to get Realms. Seems to work fine. What behaviour are you observing?

Reply:
Sorry, what I thought was committed yesterday actually wasn't. Should be up-to-date now. The issue is that if I use the service account in the unit test to list the clients, it lacks authorization. The same test works if I use a username / password.

Reply:
Looks like the assigned roles are missing from the access token when using this client. Investigating further.

Reply:
Sent https://github.com/shawkins/keycloak/pull/12 with a fix. @shawkins Please take a look when you get a chance.

Reply:
@vmuzikar thanks, that makes sense